src/app/ResourceManager/LocalizationHelpers.cs

```diff
@@ -126,62 +120,53 @@ public static string ProcessSubmoduleStatus(IGitModule module, GitSubmoduleStatu
 
             IGitModule gitModule = moduleIsParent ? module.GetSubmodule(status.Name) : module;
             StringBuilder sb = new();
-            sb.AppendLine("Submodule " + status.Name + " Change");
+            sb.AppendLine($"Submodule {status.Name}");
 
             // TEMP, will be moved in the follow up refactor
             ICommitDataManager commitDataManager = new CommitDataManager(() => gitModule);
 
             CommitData? oldCommitData = null;
-            if (status.OldCommit != status.Commit)
+            if (status.OldCommit != status.Commit && status.OldCommit is not null)
             {
                 sb.AppendLine();
-                sb.AppendLine("From:\t" + (status.OldCommit?.ToString() ?? "null"));
+                sb.Append("From:\t");
+                sb.AppendLine(status.OldCommit?.ToString() ?? "null");
 
                 // Submodule directory must exist to run commands, unknown otherwise
-                if (gitModule.IsValidGitWorkingDir())
+                if (gitModule.IsValidGitWorkingDir()
+                    && commitDataManager.GetCommitData(status.OldCommit.ToString(), cache: true) is CommitData c)
                 {
-                    if (status.OldCommit is not null)
-                    {
-                        oldCommitData = commitDataManager.GetCommitData(status.OldCommit.ToString(), cache: true);
-                    }
+                    oldCommitData = commitDataManager.GetCommitData(status.OldCommit.ToString(), cache: true);
+                    oldCommitData = c;
 
-                    if (oldCommitData is not null)
+                    sb.AppendLine("\t\t\t" + GetRelativeDateString(DateTime.UtcNow, oldCommitData.CommitDate.UtcDateTime) + " (" +
+                                  GetFullDateString(oldCommitData.CommitDate) + ")");
+                    string[] lines = oldCommitData.Body.Trim(Delimiters.LineFeedAndCarriageReturn).Split(Delimiters.LineFeedAndCarriageReturn, StringSplitOptions.None);
+                    foreach (string line in lines)
                     {
-                        sb.AppendLine("\t\t\t\t\t" + GetRelativeDateString(DateTime.UtcNow, oldCommitData.CommitDate.UtcDateTime) + " (" +
-                                      GetFullDateString(oldCommitData.CommitDate) + ")");
-                        string[] lines = oldCommitData.Body.Trim(Delimiters.LineFeedAndCarriageReturn).Split("\r\n", StringSplitOptions.None);
```

Comment:
`Split(String? separator, StringSplitOptions options = StringSplitOptions.None)`
and
`Split(String[]? separator, int count, StringSplitOptions options)`
behave differently.